Automattic/wp-calypso
# Normalize query key structure for email management react-query hooks

#### Proposed Changes

This is a step in changing the add/remove operations for email forwards so that they are implemented as react-query mutations instead of as Redux actions. Why would we want that? Because we are already using react-query queries for data fetching in the email management views, but also still using Redux for add/remove operations. This means that we don't get optimistic updates and we run a greater risk of missing data invalidations when users change something (see #64154). 

In this PR, I have made a few changes to the existing react-query wrapper hooks related to email management:

- Normalized the way we generate query keys. These keys need to be referenced from the mutations, so it's valuable to have a predictable way of generating them.
- Added `select` functions that are passed to react-query. This makes it easier to reference the data that they fetch and to default to empty arrays/objects.
- Converted the hooks to Typescript.

#### Testing Instructions



This PR has no visible changes, so testing only involves ensuring that the email fowards management views still function normally.

---

1. Have a domain with email forwards
2. Navigate to `/email/:domain/manage/:site_slug`
3. Ensure that the email forwards are loaded and displayed

---

1. Have a domain with Titan/Google Workspace
2. Navigate to `/inbox/:site_slug`
3. Ensure that the mailboxes are loaded and displayed

---

1. Have a domain without any email solution
2. Navigate to `/domains/manage/:domain/edit/:site_slug`
3. Ensure that the `DomainEmailInfoCard` is properly loaded in the right column and displays a "Add professional email" CTA

---



Related to #64447 and #64154
